colossus-metrics/src/main/scala/colossus/metrics/Collection.scala

```diff
@@ -97,18 +97,22 @@ private[metrics] class CollectionMap[T] {
 
 class DuplicateMetricException(message: String) extends Exception(message)
 
-class Collection(val config: CollectorConfig) {
-
-  val collectors : ConcurrentHashMap[MetricAddress, Collector] = new ConcurrentHashMap[MetricAddress, Collector]()
+class Collection(val config: CollectorConfig)
+  extends mutable.Map[MetricAddress, Collector]
+  with mutable.MapLike[MetricAddress, Collector, Collection] {
 
+  val suffixes: ConcurrentHashMap[MetricAddress, Collection] = new ConcurrentHashMap[MetricAddress, Collection]()
+  var collector: Option[Collector] = None
```

Comment:
If I understand correctly, this needs to be a var because you could create a metric called `/a/b/c`, which creates a 3-node trie.  Then you create a metric `/a/b` so we need to update the `b`node with a collector.  This...seems ok to me except that this has to be an `AtomicReference` since right now it's not threadsafe.  Also this should definitely not be a public var.